src/wallet/Wallet.ts

```diff
@@ -102,28 +103,29 @@ class Wallet {
     logger.log('info', `Getting transactions for ${addresses.length} addresses.`);
     const addressesWithTx: string[] = [];
     const txResults = await Promise.all(addresses.map((address) => api.getTransactions(address)));
-
-    const blockHashes: Set<string> = new Set();
     addresses.forEach((address, i) => {
       const { transactions } = txResults[i];
       logger.log('info', `${address}: ${transactions.length} transactions found.`);
       if (transactions.length !== 0) {
         const confirmedTx = transactions.filter((tx) => tx.confirmations > 0);
-        confirmedTx.forEach((tx) => blockHashes.add(tx.acceptingBlockHash));
         this.transactionsStorage[address] = confirmedTx;
         addressesWithTx.push(address);
       }
     });
-    let blockRes = await Promise.all(Array.from(blockHashes).map((hash) => api.getBlock(hash)));
-    let blockTimestamps = blockRes.flat().reduce((map: Record<string, number>, val) => {
-      map[val.blockHash] = val.timestamp;
-      return map;
-    }, {});
-    this.transactions = txParser(
-      this.transactionsStorage,
-      Object.keys(this.addressManager.all),
-      blockTimestamps
-    );
+    this.transactions = txParser(this.transactionsStorage, Object.keys(this.addressManager.all));
+    const pendingTxHashes = Object.keys(this.pending.transactions);
+    if (pendingTxHashes.length) {
+      pendingTxHashes.forEach((hash) => {
+        if (this.transactions.map((tx) => tx.transactionHash).includes(hash)) {
+          this.deletePendingTx(hash);
+        }
+      });
+    }
+    if (
+      this.transactionsStorage[this.addressManager.receiveAddress.current.address] !== undefined
```

Comment:
What do you think about pulling this out into a named boolean variable defined above? Solves the formatting issue and makes the code a bit more readable, because I'm not sure what this check is supposed to be telling us.

Maybe something like

```javascript
const receiveAddrHaxTx = this.transactionsStorage[this.addressManager.receiveAddress.current.address] !== undefined;

if (receiveAddrHasTx) {
    this.addressManager.receiveAddress.next();
}
```

Reply:
Thanks Ben, updated!